### PR TITLE
Rayyan zaid/profile info component

### PR DESCRIPTION
**Pull request by RayyanZaid**
Final changes after pull req feedback
    * Made font sizes, widths, heights, etc relative to screen sizes
    * Added an id and onclick to button
    * Changed to paragraph tags instead of h1's
    * Added function to dynamically shrink text size if it renders out of the div
        if someone has a really long email or name

ProfileInfoComp

### CODE REVIEW

File: components/Profile.jsx [hunk 1]
```diff
@@ -1,7 +1,40 @@
 import React from "react";
 
-const Profile = () => {
-  return <div>Profile</div>;
+const Profile = ({ name, major, classOf, netId, email, points }) => {
+  return (
+    <div className="grid grid-cols-2 w-[450px] p-[142px]">
+      <div>
+        <h1 className="text-[36px] h-fit w-fit font-bold">name:</h1>
```
Comment (shahdivyank) [inline components/Profile.jsx:7]: use p tags

Comment (kninja99) [reply components/Profile.jsx:7]: I'm a little confused, in another commit I saw that we want to be using react grid elements rather than css grid styles. Not sure if that still applies here but might also be something to look into.

I also would use a p tag at least for where we are injecting values using props.

File: components/Profile.jsx [hunk 2]
```diff
@@ -1,7 +1,40 @@
...
+        <h1 className="text-[30px] h-min w-[370px] pb-[32px] 30">{name}</h1>
+      </div>
+      <div className="pl-[300px]">
+        <h1 className="text-[36px] h-fit w-fit font-bold">netid:</h1>
```
Comment (kninja99) [inline components/Profile.jsx:11]: Probably can use a h3-h5 tag rather than a h1 tag

File: components/Profile.jsx [hunk 3]
```diff
@@ -1,7 +1,40 @@
...
+        <h1 className="text-[30px] h-min w-[370px] pb-[32px] 30">{netId}</h1>
+      </div>
+
+      <div>
+        <h1 className="text-[36px] h-fit w-fit font-bold">major:</h1>
+        <h1 className="text-[30px] h-min w-[370px] pb-[32px] 30">{major}</h1>
```
Comment (kninja99) [inline components/Profile.jsx:17]: switch to p tag

File: components/Profile.jsx [hunk 4]
```diff
@@ -1,7 +1,40 @@
...
+      </div>
+      <div className="pl-[300px]">
+        <h1 className="text-[36px] h-fit w-fit font-bold">email:</h1>
+        <h1 className="text-[30px] h-min w-[370px] pb-[32px] 30">{email}</h1>
```
Comment (kninja99) [inline components/Profile.jsx:21]: switch to p tag

File: components/Profile.jsx [hunk 5]
```diff
@@ -1,7 +1,40 @@
...
+      </div>
+
+      <div>
+        <h1 className="text-[36px] h-fit w-max font-bold">class of:</h1>
+        <h1 className="text-[30px] h-min w-[370px] pb-[32px] 30">{classOf}</h1>
```
Comment (kninja99) [inline components/Profile.jsx:26]: switch to p tag

File: components/Profile.jsx [hunk 6]
```diff
@@ -1,7 +1,40 @@
...
+      </div>
+      <div className="pl-[300px]">
+        <h1 className="text-[36px] h-fit w-fit font-bold">points:</h1>
+        <h1 className="text-[30px] h-min w-[370px] pb-[32px] 30">{points}</h1>
```
Comment (kninja99) [inline components/Profile.jsx:30]: switch to p tag

File: components/Profile.jsx [hunk 7]
```diff
@@ -1,7 +1,40 @@
...
+      </div>
+
+      <button class="h-[94px] w-[359px] px-6 m-2 text-[36px] font-bold text-acm-black transition-colors duration-150 bg-[#C693EA] rounded-lg focus:shadow-outline hover:bg-indigo-500">
+        edit profile
+      </button>
```
Comment (kninja99) [inline components/Profile.jsx:35]: possibly add a id to the button so we can easily target the buttons event in the future. I think currently the implementation of the button is fine since there is only one on the page that I can see but might still be good to have. 

I have not checked responsiveness on this e\component but see a lot of hard coded height and width values so we also got to make sure that it will react well with other screen sizes. Just something to keep in mind

**Review by kninja99 — CHANGES_REQUESTED** [review]

File: components/Profile.jsx (hunk 1 above)
Comment (shahdivyank) [reply components/Profile.jsx:7]: Yea we want to be using Bootstrap grid

File: components/Profile.jsx (hunk 7 above)
Comment (RayyanZaid) [reply components/Profile.jsx:35]: Thank you for all the feedback!

File: components/Profile.jsx [hunk 8]
```diff
@@ -1,7 +1,72 @@
...
+          <p className="text-[1.5vw] font-lexend h-[8vh] w-1/50 pb-[1.0vw] 30">
+            {netId}
+          </p>
+
+          <p className="text-[2.0vw] font-lexend h-fit w-fit font-bold">
+            email:
+          </p>
+          <p className={alterFontSize(email)}>{email}</p>
+
+          <p className="text-[2.0vw] font-lexend h-fit w-fit font-bold">
```
Comment (shahdivyank) [inline components/Profile.jsx:52]: use the tailwind classes

File: components/Profile.jsx [hunk 9]
```diff
@@ -1,7 +1,72 @@
...
+            points:
+          </p>
+          <p className="text-[1.5vw] font-lexend h-[8vh] w-1/50 pb-[1.0vw] 30">
```
Comment (shahdivyank) [inline components/Profile.jsx:55]: same here

File: components/Profile.jsx [hunk 10]
```diff
@@ -1,7 +1,72 @@
...
+            {points}
+          </p>
+        </Col>
+      </Row>
+
+      <button
+        id="editProfile"
+        className=" mt-[2vh] px-3 py-2 text-[1.7vw] h-[9vh] w-[20vw] font-lexend font-bold text-acm-black transition-colors duration-150 bg-[#C693EA] rounded-lg focus:shadow-outline hover:bg-indigo-500"
```
Comment (shahdivyank) [inline components/Profile.jsx:63]: same here

File: components/Profile.jsx [hunk 11]
```diff
@@ -1,7 +1,72 @@
...
+        onClick={() => console.log(" 'Edit Profile' Button clicked")}
```
Comment (shahdivyank) [inline components/Profile.jsx:64]: make a separate function at the top for this since we r gonna be doing way more things for it

File: components/Profile.jsx [hunk 12]
```diff
@@ -1,7 +1,72 @@
...
+};
+const Profile = ({ name, major, classOf, netId, email, points }) => {
+  return (
+    <div>
+      <Row className="w-3/5">
+        <Col className="w-2/3">
+          <p className="text-[2.0vw] font-lexend h-fit w-fit font-bold">
+            name:
+          </p>
+          <p className={alterFontSize(name)}>{name}</p>
```
Comment (shahdivyank) [inline components/Profile.jsx:23]: put the function directly into here

File: components/Profile.jsx [hunk 13]
```diff
@@ -1,7 +1,72 @@
...
+const alterFontSize = (word) => {
+  const numChars = word.length;
+
+  if (numChars > 40) {
+    return "text-[0.75vw] font-lexend h-[8vh] w-1/50 pb-[1.0vw] 0.5vw";
+  } else if (numChars > 35) {
+    return "text-[1.30vw] font-lexend h-[8vh] w-1/50 pb-[1.0vw] 0.5vw";
+  }
+
+  return "text-[1.5vw] font-lexend h-[8vh] w-1/50 pb-[1.0vw] 1.0vw";
```
Comment (shahdivyank) [inline components/Profile.jsx:13]: tailwind

File: components/Home.jsx [hunk 14]
```diff
@@ -1,25 +1,7 @@
 import React from "react";
 
-const Home = ({ name, points, year }) => {
```
Comment (shahdivyank) [inline components/Home.jsx:3]: Why is this deleted?

File: components/Home.jsx [hunk 15]
```diff
@@ -1,25 +1,7 @@
...
-  return (
-    <div className="ml-12 mt-20">
-      <div className="mb-4">
-        <h1 className="text-8xl text-black font-bold h-min w-min">welcome‚</h1>
-        <h1 className="text-8xl text-black font-bold">{name}</h1>
-      </div>
-
-      <div>
-        <h5 className="text-4xl text-gray-500 text-acm-gray font-acm ">
```
Comment (shahdivyank) [inline components/Home.jsx:12]: font-lexend

File: components/Profile.jsx [hunk 16]
```diff
@@ -1,7 +1,65 @@
...
+          <p className="text-acm-black text-2xl font-lexend h-fit w-fit font-bold">
+            {points}
+          </p>
+        </Col>
+      </Row>
+
+      <button
+        id="editProfile"
+        className=" mt-8 h-16 w-1/4 font-lexend font-bold text-acm-black text-2xl transition-colors duration-150 bg-acm-lightpurple rounded-lg focus:shadow-outline hover:bg-indigo-500"
+        onClick={editProfileButtonFunction}
```
Comment (shahdivyank) [inline components/Profile.jsx:57]: rename to editProfile
